src/functional/ffront/func_to_foast.py

```diff
@@ -327,28 +397,73 @@ def visit_Lt(self, node: ast.Lt, **kwargs) -> foast.CompareOperator:
     def visit_Eq(self, node: ast.Eq, **kwargs) -> foast.CompareOperator:
         return foast.CompareOperator.EQ
 
+    def _verify_builtin_function(self, node: ast.Call):
+        func_name = self._func_name(node)
+        func_info = getattr(fbuiltins, func_name).__gt_type__()
+        if not len(node.args) == len(func_info.args):
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{func_name}() expected {len(func_info.args)} positional arguments, {len(node.args)} given!",
+            )
+        elif unexpected_kwargs := set(k.arg for k in node.keywords) - set(func_info.kwargs):
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() got unexpected keyword arguments: {unexpected_kwargs}!",
+            )
+
+    def _verify_builtin_type_constructor(self, node: ast.Call):
+        if not len(node.args) == 1:
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() expected 1 positional argument, {len(node.args)} given!",
+            )
+        elif node.keywords:
+            unexpected_kwargs = set(k.arg for k in node.keywords)
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() got unexpected keyword arguments: {unexpected_kwargs}!",
+            )
+        elif not isinstance(node.args[0], ast.Constant):
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() only takes literal arguments!",
+            )
+
+    def _func_name(self, node: ast.Call) -> str:
+        func = node.func
+        if isinstance(func, ast.Name):
+            return func.id
+        return ""
```

Comment:
Everywhere you use this function you expect an `ast.Call` node with an `id` attribute of type `ast.Name`. Why not just using `node.func.id` everywhere instead of this helper function that gives you an invalid function name shifting potential errors downstream further away from the origin of the error.

Reply:
Consider this as accessing `node.func.id` with a type guard in front of the access. Mypy is happy and the pattern can get reused so I am happy too.

Does it have to get revised if we ever allow arbitrary expressions to be called? Sure.

Reply:
I will say this though. I think directly accessing `node.func.id` and letting it fail when the attribute is not there is better and cleaner code. Unfortunately mypy does not let me do it.